x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/TimeSeriesIT.java

```diff
@@ -37,6 +45,48 @@ public void testEmpty() {
                 "type=long,time_series_metric=gauge"
             )
             .get();
-        run("FROM pods | LIMIT 1").close();
+        run("METRICS pods | LIMIT 1").close();
+    }
+
+    public void testSimpleMetrics() {
+        Settings settings = Settings.builder().put("mode", "time_series").putList("routing_path", List.of("pod")).build();
+        client().admin()
+            .indices()
+            .prepareCreate("pods")
+            .setSettings(settings)
+            .setMapping(
+                "@timestamp",
+                "type=date",
+                "pod",
+                "type=keyword,time_series_dimension=true",
+                "cpu",
+                "type=double,time_series_metric=gauge"
+            )
+            .get();
+        List<String> pods = List.of("p1", "p2", "p3");
+        long startTime = DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis("2024-04-15T00:00:00Z");
+        int numDocs = between(10, 10);
+        Map<String, List<Integer>> cpus = new HashMap<>();
+        for (int i = 0; i < numDocs; i++) {
+            String pod = randomFrom(pods);
+            int cpu = randomIntBetween(0, 100);
+            cpus.computeIfAbsent(pod, k -> new ArrayList<>()).add(cpu);
+            long timestamp = startTime + (1000L * i);
+            client().prepareIndex("pods").setSource("@timestamp", timestamp, "pod", pod, "cpu", cpu).get();
+        }
+        List<String> sortedGroups = cpus.keySet().stream().sorted().toList();
+        client().admin().indices().prepareRefresh("pods").get();
+        try (EsqlQueryResponse resp = run("METRICS pods STATS load=avg(cpu) BY pod | SORT pod")) {
```

Comment:
Shouldn't the `METRICS` command at least for now be protected by tsdb query pragma?

Reply:
The metrics command is only available in snapshot builds. I'm not sure whether we should require the time-series pragmas for it. It's easier to try it without the pragmas.

Reply:
Agreed, we should start moving away from the pragma once we have a metrics cmd.

Reply:
I missed that the metrics command is only available in snapshot build. Only making it available in snapshot builds is good. The reason I mentioned this, as far as I understand the METRICS command isn't wired up yet in this PR to do time series aggregation. Additional changes are needed to do this.